Add Deploy 13.3.0 and 14.2.0 release notes and document ImportOnStartup and ValidateDependenciesOnImport settings (#6699)
umbraco/UmbracoDocs: 13/umbraco-deploy/deployment-workflow/import-on-startup.md

---
meta.Title: Import on startup
description: How to import content and schema on startup and implement your own `IArtifactImportOnStartupProvider`
---

# Import on startup

> Deploy can import content and/or schema previously exported from another Umbraco installation on start-up. This can be used to quickly setup a baseline/starter kit or as a workaround for large ZIP archives that can't be (easily) uploaded via the backoffice.

Reviewer: ```suggestion
Deploy can import content and/or schema previously exported from another Umbraco installation on start-up. This allows for a quick setup of a baseline/starter kit or serves as a workaround for large ZIP archives that cannot be uploaded via the backoffice.
```

## Default configuration

> The default configuration will look for the ZIP archive `umbraco\Deploy\import-on-startup.zip` on start-up and if it exists, run an import and delete the file on successful completion. If you want to [customize the default behaviour you can do so via settings](../getting-started/deploy-settings.md#import-on-startup).

Reviewer: ```suggestion
The default configuration will look for the ZIP archive `umbraco\Deploy\import-on-startup.zip` on start-up and if it exists, run an import and delete the file on successful completion. If you want to customize the default behavior, do so via [settings](../getting-started/deploy-settings.md#import-on-startup).
```

> This feature is extensible via a provider based model by implementing `IArtifactImportOnStartupProvider` and registering it using `builder.DeployArtifactImportOnStartupProviders()`. The default `Umbraco.Deploy.Infrastructure.SettingsArtifactImportOnStartupProvider` implementation uses the above settings and inherits from `Umbraco.Deploy.Infrastructure.ArtifactImportOnStartupProviderZipArchiveBase` (which can be used for your own custom implementation).

Reviewer: ```suggestion
This feature is extensible via a provider-based model by implementing `IArtifactImportOnStartupProvider` and registering it using `builder.DeployArtifactImportOnStartupProviders()`. The default `Umbraco.Deploy.Infrastructure.SettingsArtifactImportOnStartupProvider` implementation uses the above settings and inherits from `Umbraco.Deploy.Infrastructure.ArtifactImportOnStartupProviderZipArchiveBase` (which can be used for your own custom implementation).
```

## Implementing your own `IArtifactImportOnStartupProvider`

> An example of an import on start-up provider that imports from a physical directory (instead of ZIP archive) is shown below:

Reviewer: ```suggestion
An example of an import on a start-up provider that imports from a physical directory (instead of ZIP archive) is shown below:
```
